builder/processor/src/main/java/io/helidon/builder/processor/CustomMethods.java

```diff
@@ -0,0 +1,284 @@
+                VOID_TYPE.equals(customMethod.returnType) ? "" : "return "
+                        + customMethodsType.genericTypeName().fqName() + "."
+                        + customMethod.name()
+                        + "("
+                        + customMethod.arguments().stream().map(Argument::name).collect(Collectors.joining(", "))
+                        + ")";
+
+        // factory methods are just copied to the generated prototype
+        return new GeneratedMethod(new Method(typeInformation.prototype(),
+                                              customMethod.name(),
+                                              customMethod.returnType(),
+                                              customMethod.arguments(),
+                                              customMethod.javadoc()),
+                                   annotations,
+                                   generatedCall);
+    }
+
+    private static List<CustomMethod> findMethods(TypeContext.TypeInformation typeInformation,
+                                                  TypeInfo customMethodsType,
+                                                  Errors.Collector errors,
+                                                  TypeName requiredAnnotation,
+                                                  MethodProcessor methodProcessor) {
+        // all custom methods must be static
+        // parameter and return type validation is to be done by method processor
+        return customMethodsType.elementInfo()
+                .stream()
+                .filter(TypeInfoPredicates::isMethod)
+                .filter(TypeInfoPredicates::isStatic)
+                .filter(TypeInfoPredicates.hasAnnotation(requiredAnnotation))
+                .map(it -> {
+                    // return type
+                    TypeName returnType = it.typeName();
+                    // method name
+                    String methodName = it.elementName();
+                    // parameters
+                    List<Argument> arguments = it.parameterArguments()
+                            .stream()
+                            .map(arg -> new Argument(arg.elementName(), arg.typeName()))
+                            .toList();
+
+                    // javadoc, if present
+                    List<String> javadoc = it.description()
+                            .map(String::trim)
+                            .stream()
+                            .filter(Predicate.not(String::isBlank))
+                            .findAny()
+                            .map(description -> description.split("\n"))
+                            .map(List::of)
+                            .orElseGet(List::of);
+
+                    // annotations to be added to generated code
+                    List<String> annotations = it.findAnnotation(Types.PROTOTYPE_ANNOTATED_TYPE)
+                            .flatMap(Annotation::value)
+                            .map(annotation -> annotation.split(","))
```

Comment:
map to trim() and filter on not blank.

Reply:
I can understand the `trim` (if somebody uses a space after a comma). Not sure I understand filtering out blanks, as that is a wrong declaration (e.g. `myAnnotation,`)

Reply:
I have done both in the code, still seems a bit over-protective